privaterelay/middleware.py

```diff
@@ -12,6 +13,23 @@
 metrics = markus.get_metrics("fx-private-relay")
 
 
+class StoreUtmsInCookie:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # Store utm_ values in cookie so we can send them in GA pings after redirecting
+        # thru FxA flow.
+        response = self.get_response(request)
+        if "utm_" in request.get_full_path():
+            parsed_url = urlparse(request.build_absolute_uri())
+            parsed_qs = parse_qs(parsed_url.query)
+            for param in parsed_qs:
+                if param.startswith("utm_"):
+                    response.set_cookie(param, parsed_qs[param][0])
```

Comment:
(suggestion, blocking) Maximum size of the cookie differs from browser to browser. Do we want to set a limit on how big the cookies can get here?

Reply:
Truncating the value to 4095 bytes.